tqdm/_tqdm.py

```diff
@@ -402,8 +402,14 @@ def inner(df, func, *args, **kwargs):
                 # Precompute total iterations
                 total = getattr(df, 'ngroups', None)
                 if total is None:  # not grouped
-                    total = len(df) if isinstance(df, Series) \
-                        else df.size // len(df)
+                    if isinstance(df, Series):
+                        total = len(df)
+                    elif kwargs.get('axis', 0) == 0:
+                        # over columns
+                        total = df.size // len(df)
+                    else:
+                        # over rows
```

Comment:
Actually we can probably already support panels, it's just that @casperdcl added pandas features per user requests. Panels are quite similar to DataFrame so it should work as-is I guess.